invenio_search/__init__.py

```diff
@@ -22,7 +22,257 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-"""Invenio module for information retrieval."""
+r"""Invenio module for information retrieval.
+
+Allows retrieving records from a configurable backend (currently, from
+Elasticsearch).
+
+
+Initialization
+--------------
+
+To be able to retrieve information from *somewhere*, we first need to setup
+this *somewhere*. So make sure you have the correct version of Elasticsearch
+installed and running (currently, version 2.x and 5.x are supported):
+
+.. code-block:: console
+
+   $ elasticsearch --version
+   Version: 2.4.6, Build: 5376dca/2017-07-18T12:17:44Z, JVM: 1.8.0_144
+
+In this case, we are using version 2 of Elasticsearch, so make sure to pin the
+correct version of ``elasticseach`` and ``elasticsearch-dsl`` packages -
+simply put those 2 lines in your ``setup.py`` :
+
+.. code-block:: python
+
+    # setup.py
+    # ...
+    install_requires = [
+        ...
+        'elasticsearch>=2.0.0,<3.0.0',
+        'elasticsearch-dsl>=2.0.0,<3.0.0',
+        ...
+    ]
+
+.. _creating_index:
+
+Creating index
+~~~~~~~~~~~~~~
+
+To be able to run flask CLI commands later, we first need to have a Flask app,
+so create the following ``app.py`` file:
+
+.. code-block:: python
+
+    # app.py
+    from flask import Flask
+    from invenio_search import InvenioSearch
+
+    app = Flask('myapp')
+    search = InvenioSearch(app)
+
+This will create an empty index, which is not very useful, so let's add some
+mappings. To not reinvent the wheel, let's reuse the mappings from the example
+application. Add the following line at the end of ``app.py`` file that you
+have just created:
+
+.. code-block:: python
+
+    # app.py
+    ...
+    search.register_mappings('demo', 'examples.data.elastic_v2')
```

Comment:
It's not so clear where the mapping files coming from. Perhaps just mentioning something like.... "Above looks into directory x,y, z and disocvers files a,b, cwhich are elasticsearch mappings"